frame/contracts/src/exec.rs

```diff
@@ -80,11 +79,11 @@ pub type ExecResult = Result<ExecReturnValue, ExecError>;
 /// ownership of buffer unless there is an error.
 #[macro_export]
 macro_rules! try_or_exec_error {
-	($e:expr, $buffer:expr) => {
+	($e:expr) => {
 		match $e {
 			Ok(val) => val,
 			Err(reason) => return Err(
-				$crate::exec::ExecError { reason: reason.into(), buffer: $buffer }
+				$crate::exec::ExecError { reason: reason.into() }
```

Comment:
Removing the scratch buffer makes `ExecError` and this macro kind of useless, right? The comment is definitely needs adjustments.

Reply:
You are right. I remove the `ExecError` and replaced it all by some nice `?` syntax.